src/test/java/us/ihmc/euclid/shape/convexPolytope/ConvexPolytope3DTest.java

```diff
@@ -1906,6 +1946,215 @@ else if (convexPolytope3D.getNumberOfVertices() == 2)
+         actualSecondInstersection.setToZero();
+         assertEquals(2, convexPolytope3D.intersectionWith(pointOnFace, direction, actualFirstInstersection, actualSecondInstersection));
+
+      }
+
+      //0 intersection
+      for (int i = 0; i < ITERATIONS; i++)
+      {
+
+         ConvexPolytope3D convexPolytope3D = EuclidShapeRandomTools.nextConvexPolytope3D(random);
+
+         //Generate a point at a random place on a random face
+
+         int index = random.nextInt(convexPolytope3D.getNumberOfFaces());
+
+         Face3D face = convexPolytope3D.getFaces().get(index);
+
+         List<Vertex3D[]> triangles = new ArrayList<>();
+
+         if (convexPolytope3D.getNumberOfVertices() == 3)
+         {
+            Vertex3D[] triangle = {face.getVertex(0), face.getVertex(1), face.getVertex(2)};
+            triangles.add(triangle);
+         }
+         else
+         {
+            for (int j = 0; j < face.getVertices().size() - 2; j++)
+            {
+
+               Vertex3D[] triangle = {face.getVertex(j), face.getVertex(j + 1), face.getVertex(j + 2)};
+               triangles.add(triangle);
+            }
+         }
+
+         int randomTriangleIndex = random.nextInt(face.getVertices().size() - 2);
+
+         Vertex3D A = face.getVertex(randomTriangleIndex);
+         Vertex3D B = face.getVertex(randomTriangleIndex + 1);
+         Vertex3D C = face.getVertex(randomTriangleIndex + 2);
+
+         double alpha = random.nextDouble();
+         double beta = random.nextDouble() * (1 - alpha);
+         double gamma = 1 - alpha - beta;
+
+         double x = alpha * A.getX() + beta * B.getX() + gamma * C.getX();
+         double y = alpha * A.getY() + beta * B.getY() + gamma * C.getY();
+         double z = alpha * A.getZ() + beta * B.getZ() + gamma * C.getZ();
+
+         Point3D pointOnFace = new Point3D(x, y, z);
+
+         //Get the point outside of the face 
+
+         Vector3D direction = face.getNormal();
+         Point3D pointOutside = new Point3D();
+         pointOutside.scaleAdd(EuclidCoreRandomTools.nextDouble(random, 1.0, 10.0), direction, pointOnFace);
+
+         //Generate an orthogonal vector
+
+         Vector3D lineDirection = EuclidCoreRandomTools.nextOrthogonalVector3D(random, direction, false);
+
+         Point3D actualFirstInstersection = new Point3D();
+         Point3D actualSecondInstersection = new Point3D();
+
+         assertEquals(0, convexPolytope3D.intersectionWith(pointOutside, lineDirection, null, null));
+         assertEquals(0, convexPolytope3D.intersectionWith(pointOutside, lineDirection, actualFirstInstersection, null));
+         assertEquals(0, convexPolytope3D.intersectionWith(pointOutside, lineDirection, null, actualSecondInstersection));
+         actualFirstInstersection.setToZero();
+         actualSecondInstersection.setToZero();
+         assertEquals(0, convexPolytope3D.intersectionWith(pointOutside, lineDirection, actualFirstInstersection, actualSecondInstersection));
```

Comment:
I think we can find more constructions to cover more scenarios with no intersection